### PR TITLE
Track commands that have requested the preview

### LINKED ISSUE #1441: Send and location Messages are shown as plain text for several seconds after app launch [develop]
### Description

*Type*: Bug

*Summary*: Send and location messages are shown as long text messages for several seconds after app is launched and chat is opened for the first time. After several seconds they are transformed into expected form. Sometimes it takes more than 5 seconds on my Galaxy S7 real device.
Send (Android):
screenshot_20170712-181502

Location (iOS):
img_2109

#### Expected behavior
After chat is opened all messages are shown in a familiar form/design/content for the user 
#### Actual behavior
Send message is shown as a long text message for several seconds and than transforms into usual send message

### Reproduction
video: https://drive.google.com/open?id=0Bz3t9zSg1wb7cVlSNzBFY3VIXzQ
- Open Status
- Open group chat with send message or 1-1 chat with send message

### Additional Information
* Status version:  0.9.8-67-g8e896fcd+
* Operating System: 
Real device Samsung Galaxy S7, Android 7.0
Real device iPhone 6s, iOS 10.2.1

### PR DESCRIPTION
**Pull request by retro**
This prevents the rendering of the command source before the preview is ready

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)
#1441 

### Summary:
[comment]: # (Summarise the problem and how the pull request solves it)
Send and location messages are shown as long text messages for several seconds after app is launched and chat is opened for the first time. After several seconds they are transformed into expected form.

This commit ensures that we track commands that have it's preview requested. Currently nothing will be rendered while waiting on preview, but we could add a loading icon (spinner)

### Steps to test:
- Open Status
- Open a chat with someone
- Add location
- Close Status
- Open Status
- Open the chat where you've added the location

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

**Comment by jeluard** [comment]
Do you mind adding comments for the tricky part? Not easy to understand what is going on. 
Tests would be great too.

**Comment by retro** [comment]
@jeluard Will do!

**Comment by janherich** [comment]
@retro The logic in this PR is clear to me, but still it seems like an workaround for cases where preview caching is not present/broken. Did you find out whether caching (implemented in `:request-command-preview` handler which then dispatches to `:request-command-data`)  is working correctly ?

**Review by rasom — APPROVED** [review]

**Comment by retro** [comment]
@janherich It seemed to me that the caching works correctly. I've tested this on iOS and Android, and the bug was present only on the first chat load (before the caching kicked in)

**Comment by janherich** [comment]
Thanks for the information @retro, so this PR handles specific case where nothing is cached (first time preview is requested) and jail call takes too long, gotcha.

**Review by janherich — DISMISSED** [review]

File: src/status_im/chat/specs.cljs
Comment (flexsurfer) [inline src/status_im/chat/specs.cljs:33]: could you add spec for this map, because we will add specs to all other maps later, here we use map? because it was first step of using specs

File: src/status_im/ui/screens/db.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/db.cljs:164]: why did you add :profile/profile-edit?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/chat/specs.cljs
Comment (retro) [reply src/status_im/chat/specs.cljs:33]: Sure will do

File: src/status_im/ui/screens/db.cljs
Comment (retro) [reply src/status_im/ui/screens/db.cljs:164]: Hm, I didn't add this, I think this happened when I've rebased to develop. I don't know why it's reported as my change

**Comment by janherich** [comment]
@retro While refactoring the chat-input functionality, I realised, that message should actually not be sent before preview is generated and sent back from the jail side - https://github.com/status-im/status-react/blob/develop/src/status_im/chat/events/input.cljs#L315-L321

So we should have both cases covered -
* Whenever new message is sent, preview should be received beforehand (and cached at the same time)
* For older messages, preview must be in cache

I'm little bit reluctant to approve adding new (not trivial) workaround/patch logic, until root cause of the issue is known.

**Review by janherich — CHANGES_REQUESTED** [review]
Please find the root cause of the issue

**Comment by jeluard** [comment]
Is this PR still relevant?

**Comment by retro** [comment]
@jeluard I would wait until #1709 is merged in, and than we can figure what should be done with it.

**Comment by oskarth** [comment]
@retro if this is still blocked, would you mind indicating it as such with a label and moving it to the appropriate column? So it's clear what state this PR is in.

**Comment by oskarth** [comment]
@retro Merged, this is now unblocked I believe.

**Comment by janherich** [comment]
@oskarth @retro #1709 is not merged yet, but the preview issue is already solved there, so I would still wait till it's tested and merged.

**Comment by oskarth** [comment]
My bad, confused #1709 and #1710.

**Pull request closed without merging**
